src/functional/iterator/embedded.py

```diff
@@ -143,8 +142,33 @@ def __array__(self) -> np.ndarray:
         ...
 
 
-def _is_column(v: Any) -> TypeGuard[Column]:
-    return isinstance(v, np.ndarray)
+class Column(numpy.lib.mixins.NDArrayOperatorsMixin):
+    def __init__(self, kstart: int, data: np.ndarray):
+        self.kstart = kstart
+        self.data = data
+
+    def __getitem__(self, i: int):
+        return self.data[i - self.kstart]
+
+    def tuple_get(self, i: int):
+        if self.data.dtype.names is not None:
+            return Column(self.kstart, self.data[f"f{i}"])
+        else:
+            return Column(self.kstart, self.data[i, ...])
+
+    def __setitem__(self, i: int, v: Any):
+        self.data[i - self.kstart] = v
+
+    def __array__(self, dtype=None) -> np.ndarray:
+        return self.data if dtype is None or dtype == self.data.dtype else self.data.astype(dtype)
```

Comment:
You can use `DTypeLike` from `numpy.typing` to annotate the argument.

```suggestion
    def __array__(self, dtype: Optional[DTypeLike]=None) -> np.ndarray:
        return self.data if dtype or dtype == self.data.dtype else self.data.astype(dtype)
```